system-addon/lib/HighlightsFeed.jsm

```diff
@@ -62,10 +65,16 @@ this.HighlightsFeed = class HighlightsFeed {
         continue;
       }
 
+      // If we already have the image for the card in the cache, use that
+      // immediately. Then asynchronously fetch the image (refreshes the cache).
+      const image = this.imageCache.get(page.url);
+      this.fetchImage(page.url, page.preview_image_url);
+
       // We want the page, so update various fields for UI
       Object.assign(page, {
+        image,
+        imageLoading: true, // We always have an image - fall back to a screenshot
```

Comment:
I would assume this `imageLoading` is for other sections to have just text? Let's just rename this to `hasImage` as it would seem to imply it should be `false` if it's already loaded.